packages/bsb-core/bsb/config/_attrs.py

```diff
@@ -902,7 +919,7 @@ def __set__(self, instance, value, key=None):
         else:
             setattr(instance, self.get_ref_key(), value)
             if self.should_resolve_on_set(instance):
-                if hasattr(instance, "_config_root"):  # pragma: nocover
+                if not hasattr(instance, "_config_root") or instance._config_root is None:
```

Comment:
```suggestion
                if getattr(instance, "_config_root", None) is None:
```

since you're checking it against None either way